File: Easy/1207. Unique Number of Occurrences/Solution.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer,Integer> counter = new HashMap<>();
        HashSet<Integer> checking = new HashSet<>();
        for(int num: arr){
            if(counter.containsKey(num)){
                counter.put(num,counter.get(num)+1);
            }
            else {
                counter.put(num,1);
            }
        }
        
        for(int num1: counter.values()){
            if(checking.contains(num1)){
                return false;
            }
            else {
                checking.add(num1);
            }
        }
        return true;
        
    }
}

File: Easy/1207. Unique Number of Occurrences/Solution.cs
using System;
using System.Collections.Generic;

public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        var counter = new Dictionary<int, int>();
        var checking = new HashSet<int>();
        foreach (var num in arr) {
            if (counter.ContainsKey(num)) {
                counter[num]++;
            } else {
                counter[num] = 1;
            }
        }
        
        foreach (var num1 in counter.Values) {
            if (checking.Contains(num1)) {
                return false;
            } else {
                checking.Add(num1);
            }
        }
        return true;
    }
}